### ISSUE TITLE
False-positive inconsistent overload with TypeVar and Union

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#1808]
<h3> Environment data </h3>

-   Language Server version: 2021.9.1
-   OS and version: darwin x64
-   Python version: 3.10.0rc2
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic

<h3> Expected behaviour </h3>

_no error_

<h3> Actual behaviour </h3>

```
Overloaded function implementation is not consistent with signature of overload 1
  Type "(var: T@func | int) -> (T@func | int)" cannot be assigned to type "(var: T@func) -> T@func"
    Parameter 1: type "T@func" cannot be assigned to type "T@func | int"
```

<h3> Logs </h3>

--

<h3> Code Snippet / Additional information </h3>

```python
from typing import TypeVar, overload

class X: ...
class Y: ...

T = TypeVar("T", bound=type[X])

@overload
def func(var: T) -> T:
    ...

@overload
def func(var: int) -> int:
    ...

def func(var: T | int) -> T | int:  # error
    ...


# The type is inferred correctly
reveal_type(func(X))  # Type of "func(X)" is "Type[X]"
```

**Mypy output**
```
Revealed type is "def () -> test3.X"
```

An error also occurs when changing the TypeVar definition to
```py
T = TypeVar("T", X, Y)
```

Interestingly though, this seems to be fine:
```py
T = TypeVar("T", bound=X)
```

**Comment by judej** [comment]
Thanks for the report

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202192-16-september-2021
